File: app/src/main/java/com/example/testpdf/MainActivity.java
package com.example.testpdf;

import android.app.DownloadManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.github.barteksc.pdfviewer.PDFView;
import com.github.barteksc.pdfviewer.util.FitPolicy;

import java.io.File;

public class MainActivity extends AppCompatActivity {


    private File file;
    private long downloadID;
    private BroadcastReceiver onDownloadComplete = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            //Fetching the download id received with the broadcast
            long id = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, -1);
            //Checking if the received broadcast is for our enqueued download by matching download id
            if (downloadID == id) {
                Toast.makeText(MainActivity.this, "Download Completed", Toast.LENGTH_SHORT).show();


                File f = file;
                PDFView view = findViewById(R.id.pdf_view);


                view.fromFile(file.getAbsoluteFile()/*Uri.parse("http://www.africau.edu/images/default/sample.pdf")*/)
                        .enableSwipe(true) // allows to block changing pages using swipe
                        .swipeHorizontal(false)
                        .enableDoubletap(true)
                        .enableAnnotationRendering(false) // render annotations (such as comments, colors or forms)
                        .password(null)
                        .scrollHandle(null)
                        .enableAntialiasing(true) // improve rendering a little bit on low-res screens
                        // spacing between pages in dp. To define spacing color, set view background
                        .spacing(0)
                        .pageFitPolicy(FitPolicy.WIDTH)
                        .load();

            }
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        registerReceiver(onDownloadComplete, new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));

        beginDownload();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        unregisterReceiver(onDownloadComplete);
    }

    private void beginDownload() {
        file = new File(getExternalFilesDir(null), "Dummyll");
        /*
        Create a DownloadManager.Request with all the information necessary to start the download
         */
        DownloadManager.Request request = new DownloadManager.Request(Uri.parse("http://dev.urger.tk/files/uploads/company/id_1/documents/document_146/1563462547_file.pdf"))
                .setTitle("Dummy Filey")// Title of the Download Notification
                .setDescription("Downloadingl")// Description of the Download Notification
                .setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE)// Visibility of the download Notification
                .setDestinationUri(Uri.fromFile(file))// Uri of the destination file
                .setAllowedOverMetered(true)// Set if download is allowed on Mobile network
                .setAllowedOverRoaming(true);// Set if download is allowed on roaming network
        DownloadManager downloadManager = (DownloadManager) getSystemService(DOWNLOAD_SERVICE);
        downloadID = downloadManager.enqueue(request);// enqueue puts the download request in the queue.
    }

}

File: app/src/main/java/com/example/testpdf/MainActivity.kt

import android.content.Context.DOWNLOAD_SERVICE
import android.support.v4.content.ContextCompat.getSystemService
import android.app.DownloadManager
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import java.io.File


//package com.example.testpdf
//
//import android.app.DownloadManager
//import android.content.Context
//import android.net.Uri
//import android.support.v7.app.AppCompatActivity
//import android.os.Bundle
//import android.widget.Toast
//import com.github.barteksc.pdfviewer.util.FitPolicy
//import kotlinx.android.synthetic.main.activity_main.*
//import java.io.*
//
//class MainActivity : AppCompatActivity() {
//
//	override fun onCreate(savedInstanceState: Bundle?) {
//		super.onCreate(savedInstanceState)
//		setContentView(R.layout.activity_main)
//
//	//	val uri = Uri.parse("http://www.africau.edu/images/default/sample.pdf")
//
//
//		val document = "fgfg"
//
//		//val downloadManager = getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
//
//		val uri = Uri.parse("http://www.africau.edu/images/default/sample.pdf")
//
////	/*	val request: DownloadManager.Request = DownloadManager.Request(uri)
////		request.setNotificationVisibility(Do*/wnloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
////		downloadManager.enqueue(request)
//
//		val PDFFile = File(getExternalFilesDir(null),"Dummy")
//
//
//
//		//val PDFFile = File(Uri.parse("http://www.africau.edu/images/default/sample.pdf"))
//
//		val request = DownloadManager.Request(uri)
//				.setTitle("Dummy File")// Title of the Download Notification
//				.setDescription("Downloading")// Description of the Download Notification
//				.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE)// Visibility of the download Notification
//				.setDestinationUri(Uri.fromFile(PDFFile))// Uri of the destination file
//				.setAllowedOverMetered(true)// Set if download is allowed on Mobile network
//				.setAllowedOverRoaming(true)// Set if download is allowed on roaming network
//
//
//
//		val  downloadManager = getSystemService(DOWNLOAD_SERVICE) as DownloadManager
//
//		val downloadID = downloadManager.enqueue(request)
//
//
//
//
//		pdf_view.fromFile(PDFFile)
//				.enableSwipe(true) // allows to block changing pages using swipe
//				.swipeHorizontal(true)
//				.enableDoubletap(true)
//				.onLoad {
//					Toast.makeText(this, "test", Toast.LENGTH_LONG).show()
//				}
//				.defaultPage(0)
//				.enableAnnotationRendering(false) // render annotations (such as comments, colors or forms)
//				.password(null)
//				.scrollHandle(null)
//				.enableAntialiasing(true) // improve rendering a little bit on low-res screens
//				// spacing between pages in dp. To define spacing color, set view background
//				.spacing(0)
//				.pageFitPolicy(FitPolicy.WIDTH)
//				.load()
//	}
//}
